Formula/psalm.rb

```diff
@@ -0,0 +1,83 @@
+class Psalm < Formula
+  desc "PHP Static Analysis Tool"
+  homepage "https://psalm.dev"
+  url "https://github.com/vimeo/psalm/releases/download/4.8.1/psalm.phar"
+  sha256 "896246540c669c8d21e62e7c3865bc23cf1e7980caa920cd04b4e38a60354faf"
+  license "MIT"
+
+  depends_on "composer" => :test
+
+  uses_from_macos "php"
+
+  def install
+    bin.install "psalm.phar" => "psalm"
+  end
+
+  test do
+    (testpath/"composer.json").write <<~EOS
+      {
+        "name": "homebrew/psalm-test",
+        "description": "Testing if Psalm has been installed properly.",
+        "type": "project",
+        "require": {
+          "php": ">=7.1.3"
+        },
+        "license": "MIT",
+        "autoload": {
+          "psr-4": {
+            "Homebrew\\\\PsalmTest\\\\": "src/"
+          }
+        },
+        "minimum-stability": "stable"
+      }
+    EOS
+
+    (testpath/"src/Email.php").write <<~EOS
+      <?php
+      declare(strict_types=1);
+
+      namespace Homebrew\\PsalmTest;
+
+      final class Email
+      {
+        private string $email;
+
+        private function __construct(string $email)
+        {
+          $this->ensureIsValidEmail($email);
+
+          $this->email = $email;
+        }
+
+        public static function fromString(string $email): self
+        {
+          return new self($email);
+        }
+
+        public function __toString(): string
+        {
+          return $this->email;
+        }
+
+        private function ensureIsValidEmail(string $email): void
+        {
+          if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+            throw new \\InvalidArgumentException(
+              sprintf(
+                '"%s" is not a valid email address',
+                $email
+              )
+            );
+          }
+        }
+      }
+    EOS
+
+    system "composer", "install"
+
+    assert_match(/Config file created successfully. Please re-run psalm./,
+      shell_output("#{bin}/psalm --init"))
+    assert_match(/No errors found!/,
+      shell_output("#{bin}/psalm"))
```

Comment:
```suggestion
    assert_match "Config file created successfully. Please re-run psalm.",
                 shell_output("#{bin}/psalm --init")
    assert_match "No errors found!",
                 shell_output("#{bin}/psalm")
```
We don't need regexes here, and let's align the arguments to `assert_match` for readability.